libpod/runtime_pod_infra_linux.go

```diff
@@ -145,6 +146,34 @@ func (r *Runtime) makeInfraContainer(ctx context.Context, p *Pod, imgName, rawIm
 		if len(p.config.InfraContainer.ExitCommand) > 0 {
 			options = append(options, WithExitCommand(p.config.InfraContainer.ExitCommand))
 		}
+		if p.config.Pid.NSMode != "private" && p.config.Pid.NSMode != "" {
+			path := p.config.Pid.Value
+			ns := spec.LinuxNamespace{}
+			pidVal := p.config.Pid.NSMode
+			if pidVal != "" {
+				if pidVal == "container" {
+					err := errors.New("Invalid Argument")
+					return nil, errors.Wrap(err, "Cannot take container in a different NS as an argument")
+				}
+				if pidVal == "ns" {
+					path = p.config.Pid.Value
+					_, err := os.Stat("path")
+					if err != nil {
+						return nil, err
+					}
+					options = append(options, WithPidFile(path))
+				}
+			}
+			if pidVal != "host" {
+				ns.Type = "pid"
+				ns.Path = path
+				g.AddOrReplaceLinuxNamespace(string(ns.Type), ns.Path)
+			} else if pidVal == "host" {
+				g.Config.Linux.Namespaces = []spec.LinuxNamespace{}
```

Comment:
This wipes out *all* namespaces, not just the PID namespace. You need to loop through namespaces and make a new array containing everything except the PID namespace.